### DISCUSSION TITLE
[abctl] Unable to instal Airbyte on local widows machine

### DISCUSSION THREAD

**Discussion by Jackolas126** [airbytehq/airbyte#45358] [General]
We are currently investigating if airbyte can be used for out goals.
But we are running into a bit of a trouble when trying to instal it on a local windows machine
Multiple errors are showing and it seems that it is unable to pull the files.
We are unable to figure out what is causing the error. (also we don't have that many that are familiar with linux and docker, we are learning as we go)

>  Encountered an issue deploying Airbyte:
>             Pod: airbyte-db-0.17f3dfee3e2d446d
>             Reason: Failed
>             Message: Failed to pull image "airbyte/db:0.64.3": failed to pull and unpack image "docker.io/airbyte/db:0.64.3": failed to resolve reference "docker.io/airbyte/db:0.64.3": failed to do request: Head "https://registry-1.docker.io/v2/airbyte/db/manifests/0.64.3": tls: failed to verify certificate: x509: certificate signed by unknown authority
>             Count: 1
>   DEBUG   Encountered an issue deploying Airbyte:
>             Pod: airbyte-db-0.17f3dfee3e2f9077
>             Reason: Failed
>             Message: Error: ErrImagePull
>             Count: 1
>   DEBUG   Pulling image "airbyte/bootloader:0.64.3"
>   DEBUG   Encountered an issue deploying Airbyte:
>             Pod: airbyte-minio-0.17f3dfee417bcac6
>             Reason: Failed
>             Message: Failed to pull image "minio/minio:RELEASE.2023-11-20T22-40-07Z": failed to pull and unpack image "docker.io/minio/minio:RELEASE.2023-11-20T22-40-07Z": failed to resolve reference "docker.io/minio/minio:RELEASE.2023-11-20T22-40-07Z": failed to do request: Head "https://registry-1.docker.io/v2/minio/minio/manifests/RELEASE.2023-11-20T22-40-07Z": tls: failed to verify certificate: x509: certificate signed by unknown authority
>             Count: 1

Maybe someone can point us into the right direction
We already tried the docker by installing the "Hello world" image and that worked without any problems.

**Comment by szemek** [comment]
This looks a bit like an issue wit Docker configuration as here https://forums.docker.com/t/tls-failed-to-verify-certificate-x509/137486/9, but it might be something one layer deeper.
Do you get the same error for e.g. `docker pull airbyte/db:0.64.3` or `docker pull airbyte/bootloader:0.64.3`

**Reply by Jackolas126** [reply]
1st one looks like no problem:

>       docker pull airbyte/db:0.64.3
>       0.64.3: Pulling from airbyte/db
>       c6a83fedfae6: Pull complete
>       c0772cff1030: Pull complete
>       7cd9a10cf9bb: Pull complete
>       1f020ba7448a: Pull complete
>       e4e328cc34ca: Pull complete
>       881336ae6132: Pull complete
>       5d30650d0f2c: Pull complete
>       080a42a1ec8b: Pull complete
>       f9f6011150d4: Pull complete
>       64630d72e055: Pull complete
>       bdea2541b915: Pull complete
>       85215ea78763: Pull complete
>       Digest: sha256:792ce77375a4f85bb4718578054f20365a834d91e9f0d107bca46779c28e6ed1
>       Status: Downloaded newer image for airbyte/db:0.64.3
>       docker.io/airbyte/db:0.64.3

2nd one also seems to work without issue:

>       docker pull airbyte/bootloader:0.64.3
>       0.64.3: Pulling from airbyte/bootloader
>       0b2952a75473: Already exists
>       8891881a7cbc: Already exists
>       2eced52dfc9a: Already exists
>       60a52427e2e7: Already exists
>       47086102801e: Already exists
>       bf69eac13b2a: Already exists
>       4f4fb700ef54: Already exists
>       1c992f39b820: Pull complete
>       Digest: sha256:2a09baef745a83b7588369fa00def9a6584d1ee718f2de0072d6bae8033bbc89
>       Status: Downloaded newer image for airbyte/bootloader:0.64.3
>       docker.io/airbyte/bootloader:0.64.3

**Reply by szemek** [reply]
Okay, so it's one layer deeper, not on Docker level. It might some bug or missing edge case on Windows.
abctl uses kind cluster (https://kind.sigs.k8s.io/) under the hood (it's a kind of kubernetes in Docker)

At this moment, one workaround comes to my mind.
You can install kind executable https://kind.sigs.k8s.io/docs/user/quick-start/#installing-from-release-binaries, then you can load Docker images into
Command looks like this: `kind load docker-image ... --name airbyte-abctl`

Few weeks ago I experimented locally on macOS and I had something like this. You need to update tags to the versions used by helm chart version used by abctl.
```
kind load docker-image docker.io/airbyte/bootloader:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/connector-builder-server:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/connector-sidecar:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/cron:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/db:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/mc:latest --name airbyte-abctl
kind load docker-image docker.io/airbyte/server:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/webapp:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/worker:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/workload-api-server:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/airbyte/workload-launcher:0.63.15 --name airbyte-abctl
kind load docker-image docker.io/bitnami/kubectl:1.28.9 --name airbyte-abctl
kind load docker-image docker.io/kindest/kindnetd:v20240513-cd2ac642 --name airbyte-abctl
kind load docker-image docker.io/kindest/local-path-helper:v20230510-486859a6 --name airbyte-abctl
kind load docker-image docker.io/kindest/local-path-provisioner:v20240513-b9bba138 --name airbyte-abctl
kind load docker-image docker.io/minio/minio:RELEASE.2023-11-20T22-40-07Z --name airbyte-abctl
kind load docker-image docker.io/temporalio/auto-setup:1.23.0 --name airbyte-abctl
kind load docker-image registry.k8s.io/coredns/coredns:v1.11.1 --name airbyte-abctl
kind load docker-image registry.k8s.io/etcd:3.5.12-0 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-apiserver-arm64:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-apiserver:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-controller-manager-arm64:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-controller-manager:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-proxy-arm64:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-proxy:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-scheduler-arm64:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/kube-scheduler:v1.29.4 --name airbyte-abctl
kind load docker-image registry.k8s.io/pause:3.7 --name airbyte-abctl
```

[a comment by an automated account is not shown]
